x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -321,9 +322,9 @@ export class TaskManagerRunner implements TaskRunner {
 
       let taskParamsValidation;
       if (this.requeueInvalidTasksConfig.enabled) {
-        taskParamsValidation = this.validateTaskParams();
+        taskParamsValidation = this.validateTaskParams(modifiedContext);
```

Comment:
When adding functional tests to the task manager test suite, I noticed that the `beforeSave` and `beforeRun` middleware could potentially modify the task params which could potentially invalidate the shape of the params before this params validation occurs.

Specifically in `x-pack/test/plugin_api_integration/plugins/sample_task_plugin/server/plugin.ts`, the `beforeSave` middleware changes the task params shape while the `beforeRun` middleware restores them. In the task runner, we're validating on the task params as read from disk instead of after using `beforeRun`, so this change is to update that. I don't think it makes a difference here because we're not adding any task manager middleware but not sure what the right approach is here.

Reply:
Did we ever find anyone using the middleware?  I'm +1 on deleting it :-)

Reply:
Discussed with @mikecote and we'll keep this change for now and have opened an issue to discuss removing the middleware hooks: https://github.com/elastic/kibana/issues/163401